controllers/machinedeletionremediation_controller.go

```diff
@@ -212,6 +192,40 @@ func (r *MachineDeletionRemediationReconciler) buildMachineFromNode(node *v1.Nod
 	return machine, nil
 }
 
+func (r *MachineDeletionRemediationReconciler) verifyMachineIsDeleted(ctx context.Context, machine *unstructured.Unstructured, nodeName string) {
+	pollTime := 30 * time.Second
+	go wait.PollImmediateWithContext(ctx, pollTime, time.Hour, func(ctx context.Context) (bool, error) {
```

Comment:
Does this prevent the reconcile loop from returning for up to 1h?
If so, that seems like a problem

Reply:
It's called inside a go routine, so it returns immediately

Reply:
What's the purpose of this poll? 🤔 I think you are re-implementing / overriding the existing requeue mechanism. Why not just returning RequeueAfter 10s? Or an error for requeue with backoff?

Reply:
It's merely logging. I thought about calling requeue, but at some point the Node disappears  and I could not get its machine anymore.

Reply:
As discussed offline, it's better to store the machine name on the remediation CR. Keeping the poll alive for up to one hour is similar to saving state in memory in the reconciler, which should be avoided, because that doesn't work anymore when the operator's pod is restarted.